Remove dependency between tileset and Raster Overlay Collection

baothientran opened the pull request:
This PR is part of the refactor series #481. Raster Overlay collection now only uses `TilesetExternals` and `Tile::LoadedLinkList` to attach raster overlay to tile. This makes TilesetContentManager easier to test without involving Tileset since the manager depends on the raster overlay collection which requires tileset to be constructed in the previous implementation.

This is also the last dependency between the traversal (residing in tileset itself) and loading systems. By removing it, client can totally build their own traversal outside of Cesium Native and reuse the current loading system as they see fit

kring commented on Cesium3DTilesSelection/include/Cesium3DTilesSelection/RasterOverlayCollection.h:
```diff
@@ -1,8 +1,10 @@
-#include "Library.h"
-#include "RasterOverlay.h"
-#include "RasterOverlayTileProvider.h"
+#include <Cesium3DTilesSelection/Library.h>
```
These were using quotes before because they were relative paths to files in the same directory as this header file. The idea being that Cesium3DTilesSelection should include other parts of itself from the same directory, rather than resolving `Cesium3DTilesSelection` anew. I don't have strong feelings on this - it's pretty unlikely to make a difference to anyone - but we should try to be consistent about it.

kring commented on Cesium3DTilesSelection/src/RasterOverlayCollection.cpp:
```diff
@@ -28,14 +41,14 @@ void RasterOverlayCollection::add(std::unique_ptr<RasterOverlay>&& pOverlay) {
-      this->_pTileset->getExternals().pCreditSystem,
-      this->_pTileset->getExternals().pPrepareRendererResources,
-      this->_pTileset->getExternals().pLogger);
+      this->_externals.asyncSystem,
```
Would it be better to introduce a `RasterOverlayExternals` struct to hold these? Or perhaps just rename `TilesetExternals` to something less Tileset-specific?

kring commented on Cesium3DTilesSelection/src/RasterOverlayCollection.cpp:
```diff
@@ -8,9 +8,22 @@ using namespace CesiumGeometry;
 using namespace CesiumGeospatial;
 
 namespace Cesium3DTilesSelection {
+namespace {
+template <class Function>
```
`Tileset.h` is still included above, but I don't think it's needed anymore.

kring reviewed (commented):
More minor comments that can be addressed i the main branch.

baothientran commented:
Thanks @kring! I addressed the comment in `pr-review-integration` branch. I haven't renamed the TilesetExternals to something else though, but that can be in the future if it's needed
